adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/BrokerProxy.cs

```diff
@@ -417,7 +417,9 @@ private Bundle GetBrokerOptions(AuthenticationRequest request)
                 request.Resource);
             string computedRedirectUri = GetRedirectUriForBroker();
 
-            if (!string.Equals(computedRedirectUri, request.RedirectUri, StringComparison.OrdinalIgnoreCase))
+            //Do not remove the null or empty check from the following if statement. During the silent broker flow, there is an expectation that the
+            //redirect URI will be null. Thus, disabling this check in that scenario.
+            if (!string.IsNullOrEmpty(request.RedirectUri) && !string.Equals(computedRedirectUri, request.RedirectUri, StringComparison.OrdinalIgnoreCase))
             {
```

Comment:
this is confusing .... good to instruct devs not to remove the check. Consider making the text even simpler and possibly create a helper method which is named elegantly. 

What if the redirect uri is null, is there a bug out to fix that? is it really expected?